frontend/src/utilities/__tests__/useWatchGroups.spec.tsx

```diff
@@ -1,41 +1,112 @@
-import * as React from 'react';
-import { Provider } from 'react-redux';
-import { renderHook } from '~/__tests__/unit/testUtils/hooks';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
 import { GroupsConfig } from '~/pages/groupSettings/groupTypes';
-import { ReduxContext } from '~/redux/context';
-import { store } from '~/redux/store/store';
-import { fetchGroupsSettings } from '~/services/groupSettingsService';
+import { useAppDispatch } from '~/redux/hooks';
+import { fetchGroupsSettings, updateGroupsSettings } from '~/services/groupSettingsService';
 import { useWatchGroups } from '~/utilities/useWatchGroups';
 
 jest.mock('~/services/groupSettingsService', () => ({
   fetchGroupsSettings: jest.fn(),
+  updateGroupsSettings: jest.fn(),
+}));
+// Mock the useDispatch hook
+jest.mock('~/redux/hooks', () => ({
+  useAppDispatch: jest.fn(),
 }));
-
 const fetchGroupSettingsMock = jest.mocked(fetchGroupsSettings);
+const useAppDispatchMock = jest.mocked(useAppDispatch);
+const dispatchMock = jest.fn();
+const updateGroupSettingsMock = jest.mocked(updateGroupsSettings);
 
 describe('useWatchGroups', () => {
   it('should fetch groups successfully', async () => {
+    const mockGroupSettings: GroupsConfig = {
+      adminGroups: [{ id: 1, name: 'odh-admins', enabled: true }],
+      allowedGroups: [],
+    };
+    updateGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    useAppDispatchMock.mockReturnValueOnce(dispatchMock);
+    fetchGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    const renderResult = testHook(useWatchGroups)();
```

Comment:
May be can check the renderResult value before the waitForNextUpdate()

Reply:
Also for `expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(1) `before `waitForNextUpdate`.
May be for other test cases as well.